sw/device/silicon_creator/lib/ownership/isfb.c

```diff
@@ -0,0 +1,133 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/flash_ctrl.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/lib/manifest.h"
+#include "sw/device/silicon_creator/lib/ownership/owner_block.h"
+
+enum {
+  kFlashWordSize = 64,
+
+  // The maximum number of product expressions is 256, which maps to 512 flash
+  // words or 1024 bytes.
+  kProductExprMaxCount = 256,
+
+  // The maximum size of the product expressions in bytes.
+  KProductExprMaxCountBytes = kProductExprMaxCount * sizeof(uint32_t),
+
+  // The number of bits in the strike mask is fixed to 128. Each bit in the
+  // strike mask corresponds to a flash word.
+  kExpectedStrikeBitCount = 256,
+
+  // The expected size of the strike region in bytes and words.
+  kExpectedStrikeRegionBytesCount = kExpectedStrikeBitCount * sizeof(uint32_t),
+  kExpectedStrikeRegionWordsCount =
+      kExpectedStrikeRegionBytesCount / sizeof(uint32_t),
+};
+
+rom_error_t isfb_boot_request_process(const manifest_ext_isfb_t *ext,
+                                      const owner_config_t *owner_config,
+                                      uint32_t *checks_performed_count) {
+  *checks_performed_count = UINT32_MAX;
+  if (ext->header.name != kManifestExtIdIsfb ||
+      (hardened_bool_t)owner_config->isfb == kHardenedBoolFalse) {
```

Comment:
I tried, but I cannot find the definition of `owner_isfb_config_t` any where. Can you point it out to me please?

Also why do we cast `owner_config->isfb` to `hardened_bool_t` then later to `owner_isfb_config_t`? That doesn't seem valid.

Reply:
You can find the data structures in the ownership `datatypes.h` header in the `earlgrey_es_sival` branch. This will be merged to `earlgrey_1.0.0` once the feature is complete. 

Also take a look at `owner_config_default()` inside `owner_block.c`. Owner config entries are initialized to `kHardenedBoolFalse` to flag when an owner TLV entry is not present in the ownership blob.